### PR TITLE
drivers: sensor: ams-tmd2635: add initial driver files

### PR DESCRIPTION
**Pull request by yunheQin**
This commit adds initial driver files of ams-tmd2635

This commit modifies the KConfig and CMakeLists.txt in the "drivers/sensor" folder to add compatiable with ams-tmd2635

This commit modifies the i2c.dtsi file in the folder "tests/drivers/sensor"

### CODE REVIEW

**Review by MaureenHelm — COMMENTED** [review]
This driver looks very similar to the existing tmd2620 driver. Can you enhance the existing driver to support both instead of introducing a new separate driver?

**Comment by yunheQin** [comment]
> This driver looks very similar to the existing tmd2620 driver. Can you enhance the existing driver to support both instead of introducing a new separate driver?

Hello @MaureenHelm Firstly, thanks very much for your comments, but I assume that the it is not easy to enhance the existing driver to support both type. I investigate that the datasheet of (tmd2635 & tmd2620), there are many parameter differents, forexample the register address and the data format of the value. Secondly, I also found there are some folder like this topic. So  must I to combine these two driver files?

File: dts/bindings/sensor/ams,tmd26xx.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 Carl Zeiss Meditec AG
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    OSRAM ams TMD26XX Proximity Sensor
+
+compatible: "ams,tmd26xx"
+
+include: [sensor-device.yaml, i2c-device.yaml]
```
Comment (jilaypandya) [inline dts/bindings/sensor/ams,tmd26xx.yaml:9]: you could probably also include ams, tmd2620.yaml over here, and then you need to add only the new properties that are not already present in tmd2620, since tmd2620 properties are a subset of the tmd26xx properties.

File: drivers/sensor/tmd26xx/tmd26xx.h [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/gpio.h>
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_TMD26XX_TMD26XX_H_
+#define ZEPHYR_DRIVERS_SENSOR_TMD26XX_TMD26XX_H_
+
+#define TMD2635_CHIP_ID 0X44
```
Comment (jilaypandya) [inline drivers/sensor/tmd26xx/tmd26xx.h:13]: If possible please move the register defines to tmd26xx_reg.h. Access to these defines might come handy when developing an emul and eventually testing the driver

File: drivers/sensor/tmd26xx/tmd26xx_trigger.c [hunk 3]
```diff
@@ -1,8 +1,10 @@
 /*
- * Copyright (c) 2023 PHYTEC Messtechnik GmbH
+ * SPDX-FileCopyrightText: Copyright (c) 2023 PHYTEC Messtechnik GmbH
```
Comment (jilaypandya) [inline drivers/sensor/tmd26xx/tmd26xx_trigger.c:2]: dont modify the copyright of PHYTEC Messtechnik GmbH

File: tests/drivers/build_all/sensor/i2c.dtsi [hunk 4]
```diff
@@ -637,8 +637,8 @@ test_i2c_vl53l1x: vl53l1x@61 {
 	xshut-gpios = <&test_gpio 0 0>;
 };
 
-test_i2c_tmd2620: tmd2620@62 {
-	compatible = "ams,tmd2620";
+test_i2c_tmd26xx: tmd26xx@62 {
```
Comment (jilaypandya) [inline tests/drivers/build_all/sensor/i2c.dtsi:640]: can you revert this change

File: drivers/sensor/tmd26xx/Kconfig [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+config TMD26XX_SELECT_TMD2635
+	bool "supports tmd2635"
+	depends on TMD26XX
+	select TMD2635
+
+endchoice
+
+config TMD2635
+	bool
```
Comment (jilaypandya) [inline drivers/sensor/tmd26xx/Kconfig:51]: is this kconfig variable required? you have a choice between the two

File: drivers/sensor/tmd26xx/Kconfig [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+choice	TMD26XX_SELECT_MODE
+	prompt "Select Sensor type"
+	default TMD26XX_SELECT_NONE
+	help
+	  Specify the type of triggering to be used by the driver.
+
+config TMD26XX_SELECT_NONE
+	bool "select none default support tmd2620"
```
Comment (jilaypandya) [inline drivers/sensor/tmd26xx/Kconfig:41]: SELECT_TMD2620

File: drivers/sensor/tmd26xx/Kconfig (hunk 5 above)
Comment (yunheQin) [reply drivers/sensor/tmd26xx/Kconfig:51]: this variable is required otherwise the driver will not be initialized

File: drivers/sensor/tmd26xx/Kconfig (hunk 6 above)
Comment (yunheQin) [reply drivers/sensor/tmd26xx/Kconfig:41]: have modified refer to your proposal
